### ISSUE TITLE
NPE when checking lambda in field initializer

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#403]
```
What steps will reproduce the problem?
Compile the attached Java class using `javac -processor 
org.checkerframework.checker.nullness.NullnessChecker -version -verbose 
-AprintErrorStack -AprintAllQualifiers StringWrapper.java`

What is the expected output? What do you see instead?
I expect the compilation to complete successfully. Instead, a 
NullPointerException is thrown. See the attached compiler output.

What version of the product are you using? On what operating system?
I built the "trunk" versions of jsr308-langtools, annotation-tools and 
checker-framework from source, as described in the manual, using jdk1.8.0_31. 
I'm running linux.

Please provide any additional information below.
Didn't try to debug the issue, but the problem seems to be with the `w -> 
w.value` subexpression. The NPE is not triggered when one introduces a 
`#getValue()` accessor method and replaces the `w -> w.value` subexpression 
with a `StringWrapper::getValue` method handle reference.

```

Original issue reported on code.google.com by `stephan...@gmail.com` on 28 Feb 2015 at 10:36
